scripts/src/java/org/oppia/android/scripts/maven/GenerateMavenDependenciesList.kt

```diff
@@ -0,0 +1,351 @@
+
+    val dependenciesListFromTextproto = retrieveMavenDependencyList(pathToProtoBinary)
+
+    val updatedDependneciesList = addChangesFromTextProto(
+      dependenciesListFromPom,
+      dependenciesListFromTextproto
+    )
+
+    val manuallyUpdatedLicenses = retrieveManuallyUpdatedLicensesSet(updatedDependneciesList)
+
+    val finalDependenciesList = updateMavenDependenciesList(
+      updatedDependneciesList,
+      manuallyUpdatedLicenses
+    )
+    writeTextProto(
+      pathToMavenDependenciesTextProto,
+      MavenDependencyList.newBuilder().addAllMavenDependency(finalDependenciesList).build()
+    )
+
+    val licensesToBeFixed = getAllBrokenLicenses(finalDependenciesList)
+    if (licensesToBeFixed.isNotEmpty()) {
+      println("\nPlease verify the license link for the following licenses manually:")
+      licensesToBeFixed.forEach {
+        println("\nlicense_name: ${it.licenseName}")
+        println("original_link: ${it.originalLink}")
+        println("verified_link_case: ${it.verifiedLinkCase}")
+        println("is_original_link_invalid: ${it.isOriginalLinkInvalid}")
+      }
+      throw Exception("Licenses details are not completed")
+    }
+
+    val dependenciesWithoutAnyLinks = getDependenciesThatNeedIntervention(finalDependenciesList)
+    if (dependenciesWithoutAnyLinks.isNotEmpty()) {
+      println(
+        """
+        Please remove all the invalid links (if any) for the below mentioned dependencies
+        and provide the valid license links manually:
+        """.trimIndent()
+      )
+      dependenciesWithoutAnyLinks.forEach { dependency ->
+        println(dependency)
+      }
+      throw Exception("License links are invalid or not available for some dependencies")
+    }
+    println("\nScript executed succesfully: maven_dependencies.textproto updated successfully.")
+  }
+
+  private fun addChangesFromTextProto(
+    dependencyListFromPom: List<MavenDependency>,
+    dependencyListFromProto: List<MavenDependency>
+  ): List<MavenDependency> {
+    return dependencyListFromPom.map { dependency ->
+      dependencyListFromProto.find {
+        it.artifactName == dependency.artifactName
+      } ?: dependency
+    }
+  }
+
+  private fun retrieveManuallyUpdatedLicensesSet(
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<License> {
+    return mavenDependenciesList.flatMap { dependency ->
+      dependency.licenseList.filter { license ->
+        !license.verifiedLinkCase.equals(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET)
+      }
+    }.toSet()
+  }
+
+  private fun getDependenciesThatNeedIntervention(
+    mavenDependenciesList: List<MavenDependency>
+  ): Set<MavenDependency> {
+    return mavenDependenciesList.filter { dependency ->
+      dependency.licenseList.isEmpty() ||
+        dependency.licenseList.filter { license ->
+          license.verifiedLinkCase.equals(License.VerifiedLinkCase.VERIFIEDLINK_NOT_SET) &&
+            license.isOriginalLinkInvalid == true
```

Comment:
```suggestion
            license.isOriginalLinkInvalid
```

``isOriginalLinkInvalid`` is a boolean.

Ditto below for the false check (can just do ``!isOriginalLinkInvalid``).

Reply:
Done.